### DISCUSSION TITLE
About objects names.

### DISCUSSION THREAD

**Discussion by thomasmansencal** [colour-science/colour#694] [Ideas]
It would be great now that we have a fair bunch of nice objects and that we are still in _alpha_ to decide once for all the naming convention we want for the objects, I'm thinking especially about the ones with the author name like `colour.lightness_glasser1958` for example.

Should we prefer `colour.lightness_glasser_1958`, `colour.lightness_Glasser1958` or `colour.lightness_Glasser_1958`?

I have created a spreadsheet with the whole API public objects so that we have a broad overview and select what we prefer: 

https://docs.google.com/spreadsheets/d/1IcjF2tXptTMCwurzZ4fejI4oD7uDjOE0mxkiVw3gkqU/edit?usp=sharing

**Comment by MichaelMauderer** [comment]
I like (B) or (B) all small (for PEP8 compliance). 

I somehow perceive the name+year combo as a single unit. Having no underscores makes clear that it belongs together and the last block of the function name can always be assumed to be this identifier. And very often it will also be similar (or the same) as the reference making the link for the reader easier.

**Comment by thomasmansencal** [comment]
Yeah I had a preference for that to, I'm not that bothered with PEP8, provided it is visually "elegant" and doesn't contradicts the literature too much. I'm kind of afraid of the `xyz_to_xyy` thing.

 I'll update the spreadsheet accordingly to remove a bit of visual noise!

**Comment by MichaelMauderer** [comment]
To be honest to me `VonKries` looks a lot nicer and more readable than `vonkries`...

**Comment by thomasmansencal** [comment]
I still didn't updated the spreadsheet, hopefully tonight / tomorrow but yeah I prefer VonKries too.

**Comment by thomasmansencal** [comment]
Spreadsheet updated! Went for the 'With Respect' naming convention.

https://docs.google.com/spreadsheets/d/1IcjF2tXptTMCwurzZ4fejI4oD7uDjOE0mxkiVw3gkqU/edit?usp=sharing
